whitesands/code/game/objects/items/airlock_painter.dm

```diff
@@ -129,91 +129,158 @@
+	"corner_techfloor_grid","steel_grid","warning","warningcorner","warningfull","warningcee",
+	"warningfulltile","danger","dangercorner","dangerfull","dangercee"
+	)
+
+	var/list/decal_no_dirs = list("delivery", "warningfull", "box", "warningfulltile", "overstripefull", "stripefull", "stripefulltile", "outline","spline_fancy_full","spline_plain_full","solarpanel","traction","manydot","manydot_tiled","dangerfull","bordercolorfull","bordercolormonofull","",)
+	var/list/decal_eight_dirs = list()
+
+/obj/item/decal_painter/afterattack(var/atom/A, var/mob/user, proximity, params)
 	if(!proximity)
-		to_chat(user, "<span class='notice'>You need to get closer!</span>")
 		return
-	if(isturf(F))
-		F.AddComponent(/datum/component/decal, 'icons/turf/decals.dmi', stored_decal_total, stored_dir, CLEAN_TYPE_PAINT, color, null, null, alpha)
-		playsound(src.loc, 'sound/effects/spray2.ogg', 50, TRUE)
-
-/obj/item/decal_painter/AltClick(mob/user)
-	. = ..()
-	ui_interact(user)
-
-/obj/item/decal_painter/proc/update_decal_path()
-	var/yellow_fix = "" //This will have to do until someone refactors markings.dm
-	if (stored_color)
-		yellow_fix = "_"
-	stored_decal_total = "[stored_decal][yellow_fix][stored_color]"
-	return
-
-/obj/item/decal_painter/ui_interact(mob/user, datum/tgui/ui)
-	ui = SStgui.try_update_ui(user, src, ui)
-	if(!ui)
-		ui = new(user, src, "DecalPainter", name)
-		ui.open()
-
-/obj/item/decal_painter/ui_data(mob/user)
-	var/list/data = list()
-	data["decal_direction"] = stored_dir
-	data["decal_color"] = stored_color
-	data["decal_style"] = stored_decal
-	data["decal_list"] = list()
-	data["color_list"] = list()
-	data["dir_list"] = list()
-
-	for(var/i in decal_list)
-		data["decal_list"] += list(list(
-			"name" = i[1],
-			"decal" = i[2]
-		))
-	for(var/j in color_list)
-		data["color_list"] += list(list(
-			"colors" = j
-		))
-	for(var/k in dir_list)
-		data["dir_list"] += list(list(
-			"dirs" = k
-		))
-	return data
-
-/obj/item/decal_painter/ui_act(action,list/params)
+
+	var/turf/open/floor/F = A
+	if(!istype(F))
+		to_chat(user, "<span class='warning'>\The [src] can only be used on flooring.</span>")
+		return
+	if(color_disallowed.Find(decal_state))
+		F.AddComponent(/datum/component/decal, 'whitesands/icons/turf/decals.dmi', decal_state, decal_dir, CLEAN_TYPE_PAINT, color, null, null, alpha)
+	else
+		F.AddComponent(/datum/component/decal, 'whitesands/icons/turf/decals.dmi', decal_state, decal_dir, CLEAN_TYPE_PAINT, decal_color, null, null, alpha)
+	playsound(src.loc, 'sound/effects/spray2.ogg', 50, TRUE)
+
+/obj/item/decal_painter/attack_self(var/mob/user)
+	if(!user)
+		return FALSE
+	user.set_machine(src)
+	interact(user)
+	return TRUE
+
+/obj/item/decal_painter/interact(mob/user as mob) //TODO: Make TGUI for this because ouch
+	if(!decal_icon)
+		decal_icon = icon('whitesands/icons/turf/decals.dmi', decal_state, decal_dir)
+	user << browse_rsc(decal_icon, "floor.png")
```

Comment:
The CI is mad about this line. There are nothing you can do it. Except increase the value in the `check_regex.yaml` at the root folder.
```
  - exactly: [306, 'non-bitwise << uses', '(?<!\d)(?<!\d\s)(?<!<)<<(?!=|\s\d|\d|<|\/)']
```
to
```
  - exactly: [307, 'non-bitwise << uses', '(?<!\d)(?<!\d\s)(?<!<)<<(?!=|\s\d|\d|<|\/)']
```